java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/CodeActionsProvider.java

```diff
@@ -148,34 +136,9 @@ protected static String createLabel(CompilationInfo info, ExecutableElement e) {
 
     protected static String createLabel(CompilationInfo info, ExecutableElement e, boolean fqn) {
         StringBuilder sb = new StringBuilder();
-        if (e.getKind() == ElementKind.CONSTRUCTOR) {
-            sb.append(e.getEnclosingElement().getSimpleName());
-        } else {
-            sb.append(e.getSimpleName());
-        }
-        sb.append("("); // NOI18N
-        for (Iterator<? extends VariableElement> it = e.getParameters().iterator(); it.hasNext();) {
-            VariableElement param = it.next();
-            if (!it.hasNext() && e.isVarArgs() && param.asType().getKind() == TypeKind.ARRAY) {
-                sb.append(Utilities.getTypeName(info, ((ArrayType) param.asType()).getComponentType(), fqn));
-                sb.append("...");
-            } else {
-                sb.append(Utilities.getTypeName(info, param.asType(), fqn));
-            }
-            sb.append(" "); // NOI18N
-            sb.append(param.getSimpleName());
-            if (it.hasNext()) {
-                sb.append(", "); // NOI18N
-            }
-        }
-        sb.append(")"); // NOI18N
-        if (e.getKind() != ElementKind.CONSTRUCTOR) {
-            TypeMirror rt = e.getReturnType();
-            if (rt.getKind() != TypeKind.VOID) {
-                sb.append(" : "); // NOI18N
-                sb.append(Utilities.getTypeName(info, e.getReturnType(), fqn));
-            }
-        }
+        sb.append(Utils.label(info, e, fqn));
+        sb.append(" : "); // NOI18N
+        sb.append(Utils.detail(info, e, fqn));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 138:142, 126:130.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.